filter/ArticleMedraXmlFilter.inc.php

```diff
@@ -413,6 +466,89 @@ function createSubjectNode($doc, $subjectSchemeId, $subjectHeadingOrCode, $subje
 		return $subjectNode;
 	}
 
+	/**
+	 * Append the collection node 'collection property="crawler-based"' to the Article/Issue data node.
+	 * @param $doc DOMDocument
+	 * @param $deployment Deployment
+	 * @param $request Request
+	 * @param $context Context
+	 * @param $crawlableDataNode DOMElement
+	 * @param $article Article/Issue
+	 * @param $galleys array of galleys
+	 */
+	function appendAsCrawledCollectionNodes($doc, $deployment, $request, $context, $crawlableDataNode, $article, $galleys) {
+
+		foreach ($galleys as $crawledGalley) {
+			// loop through values 
+			// as-crawled URL - collection nodes
+		   $urlPath_ = array($article->getBestArticleId(), $crawledGalley->getBestGalleyId());
+		   $resourceURL_ = $request->url($context->getPath(), 'article', 'view', $urlPath_, null, null, true);
+		   // iParadigms crawler based collection element
+		   if($resourceURL_){
+			   $crawlerBasedCollectionNode = $doc->createElementNS($deployment->getNamespace(), 'Collection');
+			   $crawlerBasedCollectionNode->setAttribute('property', 'crawler-based');
+			   $iParadigmsItemNode = $doc->createElementNS($deployment->getNamespace(), 'Item');
+			   $iParadigmsItemNode->setAttribute('crawler', 'iParadigms');
+			   $iParadigmsItemNode->appendChild($node = $doc->createElementNS($deployment->getNamespace(), 'Resource', htmlspecialchars($resourceURL_)));
+			   $crawlerBasedCollectionNode->appendChild($iParadigmsItemNode);
+			   $crawlableDataNode->appendChild($crawlerBasedCollectionNode);
+		   }
+	   }
+	}
+
+	/**
+	 * Append the collection node 'collection property="text-mining"' to the Article/Issue data node.
+	 * @param $doc DOMDocument
+	 * @param $deployment Deployment
+	 * @param $request Request
+	 * @param $context Context
+	 * @param $textMiningCollectionDataNode DOMElement
+	 * @param $article Article
+	 * @param $galleys array of galleys
+	 */
+	function appendTextMiningCollectionNodes($doc, $deployment, $request, $context, $textMiningCollectionDataNode, $article, $galleys) {
+
+		foreach ($galleys as $submissionGalley) {
+			$textMiningCollectionNode = $doc->createElementNS($deployment->getNamespace(), 'Collection');
+			$textMiningCollectionNode->setAttribute('property', 'text-mining');
+			$urlPath__ = array($article->getBestArticleId(), $submissionGalley->getBestGalleyId());
+			$resourceURL__ = $request->url($context->getPath(), 'article', 'view', $urlPath__, null, null, true);
```

Comment:
Here too: I suppose this should be the file download URL i.e. 'download' instead of 'view', as for Crossref export/registration?

Reply:
if there is a galley with a pdf then the URL should be that of the PFD. If there isn’t any galley with pdf as the publisher publishes only the full text then use the view of the article.
Not sure whether in the URL there should always be download instead of view: see example http://www.jgerontology-geriatrics.com/article/view/493 the article at this view is published full text, but there is also the “download pdf” button that opens the URL http://www.jgerontology-geriatrics.com/article/view/493/457 (view of a galley?) but it is also downloadable as http://www.jgerontology-geriatrics.com/article/download/493/457. In any case it is ok to do as Crossref’s for consistency.